### ISSUE TITLE
CRF Layer rename sparse_targets to one_hot_target

### ISSUE THREAD

**Issue by KiddoThe2B** [farizrahman4u/keras-contrib#179]
Hi everyone,

This is a recommendation for the CRF layer, written by @linxihui . I think author's acceptance as also broad acceptance by users is needed to proceed in any code change.

I would suggest the renaming of the parameter `sparse_target`. In my opinion, it is completely misleading. One has to use either `sparse_target=False` for one-hot vectors (labels) or `sparse_target=True` to use integers. 

What is the intuition behind the definition of "sparsity" in this statement? I was completely mislead until I show the data representation of the related example (BILSTM-CRF, CONLL). 

I would suggest the clearest possible form e.g. `one_hot_target`. I would like to hear more opinion on that matter.

**Comment by tkorach** [comment]
Agreed - the description sounds contradictory to the argument name, and the documentation does not specify what the value True entails.

**Comment by WillSkywalker** [comment]
I was confused as hell for an entire day before I found this issue. So please do this!
